manifests/resource/location.pp

```diff
@@ -224,167 +223,15 @@
-  }
-  if ($fastcgi_split_path != undef) {
-    validate_string($fastcgi_split_path)
-  }
-  if ($fastcgi_index != undef) {
-    validate_string($fastcgi_index)
-  }
-  if ($uwsgi != undef) {
-    validate_string($uwsgi)
-  }
-  if ($uwsgi_param != undef) {
-    validate_hash($uwsgi_param)
-  }
-  validate_string($uwsgi_params)
-  if ($uwsgi_read_timeout != undef) {
-    validate_string($uwsgi_read_timeout)
-  }
-
-  validate_bool($internal)
-
-  validate_bool($ssl)
-  validate_bool($ssl_only)
-  if ($location_alias != undef) {
-    validate_string($location_alias)
-  }
-  if ($location_satisfy != undef) {
-    validate_re($location_satisfy, '^(any|all)$',
-    "${$location_satisfy} is not supported for location_satisfy. Allowed values are 'any' and 'all'.")
-  }
-  if ($location_allow != undef) {
-    validate_array($location_allow)
-  }
-  if ($location_deny != undef) {
-    validate_array($location_deny)
-  }
-  if ($option != undef) {
-    warning('The $option parameter has no effect and is deprecated.')
-  }
-  if ($stub_status != undef) {
-    validate_bool($stub_status)
-  }
-  if ($raw_prepend != undef) {
-    if (is_array($raw_prepend)) {
-      validate_array($raw_prepend)
-    } else {
-      validate_string($raw_prepend)
-    }
-  }
-  if ($raw_append != undef) {
-    if (is_array($raw_append)) {
-      validate_array($raw_append)
-    } else {
-      validate_string($raw_append)
-    }
-  }
-  if ($location_custom_cfg != undef) {
-    validate_hash($location_custom_cfg)
-  }
-  if ($location_cfg_prepend != undef) {
-    validate_hash($location_cfg_prepend)
-  }
-  if ($location_cfg_append != undef) {
-    validate_hash($location_cfg_append)
-  }
-  if ($include != undef) {
-    validate_array($include)
-  }
-  if ($try_files != undef) {
-    validate_array($try_files)
-  }
-  if ($proxy_cache != false) {
-    validate_string($proxy_cache)
-  }
-  if ($proxy_cache_key != undef) {
-    validate_string($proxy_cache_key)
-  }
-  if ($proxy_cache_use_stale != undef) {
-    validate_string($proxy_cache_use_stale)
-  }
-  if ($proxy_cache_valid != false) {
-    if !(is_array($proxy_cache_valid) or is_string($proxy_cache_valid)) {
-      fail('$proxy_cache_valid must be a string or an array or false.')
-    }
-  }
-  if ($proxy_method != undef) {
-    validate_string($proxy_method)
-  }
-  if ($proxy_http_version != undef) {
-    validate_string($proxy_http_version)
-  }
-  if ($proxy_set_body != undef) {
-    validate_string($proxy_set_body)
-  }
-  if ($proxy_buffering != undef) {
-    validate_re($proxy_buffering, '^(on|off)$')
-  }
-  if ($auth_basic != undef) {
-    validate_string($auth_basic)
-  }
-  if ($auth_basic_user_file != undef) {
-    validate_string($auth_basic_user_file)
-  }
-  if ($auth_request != undef) {
-    validate_string($auth_request)
-  }
-  if !is_integer($priority) {
-    fail('$priority must be an integer.')
-  }
-  validate_array($rewrite_rules)
   if (($priority + 0) < 401) or (($priority + 0) > 599) {
```

Comment:
the integer data types accepts min max params: https://docs.puppet.com/puppet/4.9/lang_data_number.html#the-integer-data-type

this conditional can be directly specified with the param type.